flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannelTest.java

```diff
@@ -269,6 +270,24 @@ public void testProducerFailedException() throws Exception {
 		ch.getNextBuffer();
 	}
 
+	/**
+	 * Tests that {@link LocalInputChannel#requestSubpartition(int)} throws {@link PartitionNotFoundException}
+	 * if the result partition was not registered in {@link ResultPartitionManager} before.
+	 */
+	@Test
+	public void testPartitionNotFoundExceptionWhileRequestingPartition() throws Exception {
+		final SingleInputGate inputGate = createSingleInputGate(1);
+		final LocalInputChannel localChannel = createLocalInputChannel(inputGate, new ResultPartitionManager());
+
+		try {
+			localChannel.requestSubpartition(0);
+
+			fail("Should throw PartitionNotFoundException while requesting unregistered partition.");
+		} catch (PartitionNotFoundException notFound) {
+			assertEquals(localChannel.getPartitionId(), notFound.getPartitionId());
+		}
+	}
```

Comment:
Maybe we could also check that the correct error is set if there is a small backoff (e.g. `10 ms`) and no partition. This is a slightly different code path.

Reply:
Yes, you are right, we should add this path.